util2.py

```diff
@@ -0,0 +1,17 @@
+def parseBoolValue(value: str | None, fail_on_errors: bool = True, preserve_none: bool = False) -> bool | None:
+    """Parses a string representing a boolean value. If parsing was successful,
+       returns `True` or `False`. If `preserve_none=True`, returns `True`, `False`,
+       or `None`. If parsing was not successful, raises `ValueError`, or, if
+       `fail_on_errors=False`, returns `None`."""
+    if not isinstance(value, str):
+        if preserve_none and value is None:
+            return value
+        return bool(value)
+    elif preserve_none and value.lower() == 'none':
+        return None
+    elif value.lower() in ('true', 'yes', 'y', 'on', '1'):
+        return True
+    elif value.lower() in ('false', 'no', 'n', 'off', '0', 'none'):
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/rkj6124/test/pull/2#issuecomment-2413706579>#b6ea43</a><span> - 10/15/2024, 05:29 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The function treats 'none' as False when preserve_none is False, which may lead to unexpected behavior. This is inconsistent with the treatment of None values for non-string inputs.
</div>
<div id="fix">
<b>Fix:</b> Remove 'none' from the list of values that return False when preserve_none is False. Instead, handle it separately to maintain consistency with None value handling.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -14,2 +14,4 @@
 -    elif value.lower() in ('false', 'no', 'n', 'off', '0', 'none'):
 -        return False
 +    elif value.lower() in ('false', 'no', 'n', 'off', '0'):
 +        return False
 +    elif value.lower() == 'none':
 +        return None if preserve_none else False
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged